project/script_IR.tcl

```diff
@@ -8,23 +8,27 @@ open_project -reset inputrouter
 
 # source files
 set CFLAGS {-std=c++11 -I../TrackletAlgorithm}
-set_top InputRouter
-add_files ../TrackletAlgorithm/InputRouter.cpp -cflags "$CFLAGS"
-add_files -tb ../TestBenches/InputRouter_test.cpp -cflags "CFLAGS"
+set_top InputRouterTop
+add_files ../TrackletAlgorithm/InputRouterTop.cc -cflags "$CFLAGS"
+add_files -tb ../TestBenches/InputRouter_test.cpp -cflags "$CFLAGS"
 
 open_solution "solution1"
 
 # Define FPGA, clock frequency & common HLS settings.
 source settings_hls.tcl
 
 # data files
-add_files -tb ../emData/IL/
+add_files -tb ../emData/
 
-csim_design -compiler gcc -mflags "-j8"
+create_clock -period 240MHz -name slow_clock 
+create_clock -period 360MHz -name fast_clock
+
+set nProc [exec nproc]
+csim_design -compiler gcc -mflags "-j$nProc"
 csynth_design
 cosim_design 
-export_design -format ip_catalog
+export_design -format ip_catalog -flow syn
```

Comment:
The other modules don't run the synthesis during export by default, i.e., during the CI tests. However, I actually like this idea. Implementation is probably much too slow for the CI tests, but synthesis still lets users get a probably-more-accurate look at the resource utilization and timing.

So I support the change you've made, and would like to see it replicated for the other modules, which we could do in a separate PR. But I wonder if @pwittich or @RobertGlein have thoughts on this?

Reply:
@aehart The -flow syn synthesizes the design again to double-check it with the results of the csynth. I think if we want to go this path we should do -flow impl. It takes longer but gives you the best estimation based on an implementation in an empty FPGA. Time it takes for the ME:
export_design -format ip_catalog # ~1 min
export_design -format ip_catalog -flow syn # ~10 min
export_design -format ip_catalog -flow impl # ~17 min